src/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/CoreTypes.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Reflection.TypeLoading
+{
+    /// <summary>
+    /// A convenience class that holds the palette of core types that were successfully loaded (or the reason they were not.)
+    /// </summary>
+    internal sealed class CoreTypes
+    {
+        private readonly RoType[] _coreTypes;
+        private readonly Exception[] _exceptions;
+
+        internal CoreTypes(MetadataLoadContext loader)
+        {
+            int numCoreTypes = (int)CoreType.NumCoreTypes;
+            RoType[] coreTypes = new RoType[numCoreTypes];
+            Exception[] exceptions = new Exception[numCoreTypes];
+            RoAssembly coreAssembly = loader.TryGetCoreAssembly(out Exception e);
+            if (coreAssembly == null)
+            {
+                // Populate exceptions[] in case there is subsquent access to the missing types
+                for (int i = 0; i < numCoreTypes; i++)
```

Comment:
This is not needed since we will throw exception for null coreAssembly a few lines below anyway. We should just throw here right away. (Move the throw statement from below up to here.)